dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/metrics/DefaultMetricsServiceExporter.java

```diff
@@ -0,0 +1,159 @@
+import org.apache.dubbo.config.ProtocolConfig;
+import org.apache.dubbo.config.RegistryConfig;
+import org.apache.dubbo.config.ServiceConfig;
+import org.apache.dubbo.rpc.model.ApplicationModel;
+import org.apache.dubbo.rpc.model.ScopeModelAware;
+
+import java.util.List;
+
+import static org.apache.dubbo.common.constants.CommonConstants.DUBBO_PROTOCOL;
+import static org.apache.dubbo.common.constants.MetricsConstants.PROTOCOL_PROMETHEUS;
+
+/**
+ * Default implementation of {@link MetricsServiceExporter}
+ */
+public class DefaultMetricsServiceExporter implements MetricsServiceExporter, ScopeModelAware {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private ApplicationModel applicationModel;
+    private MetricsService metricsService;
+    private volatile ServiceConfig<MetricsService> serviceConfig;
+
+    public DefaultMetricsServiceExporter() {
+    }
+
+    @Override
+    public void init() {
+        initialize();
+    }
+
+    private void initialize() {
+        MetricsConfig metricsConfig = applicationModel.getApplicationConfigManager().getMetrics().orElse(null);
+        // TODO compatible with old usage of metrics, remove protocol check after new metrics is ready for use.
+        if (metricsConfig != null && PROTOCOL_PROMETHEUS.equals(metricsConfig.getProtocol())) {
+            ExtensionLoader<MetricsService> extensionLoader = applicationModel.getExtensionLoader(MetricsService.class);
+            if (!extensionLoader.hasExtension(MetricsService.DEFAULT_EXTENSION_NAME)) {
+                throw new IllegalStateException("Metrics config exist, but the dubbo-metrics-api dependency is missing. Please check your project dependencies.");
+            } else {
+                this.metricsService = extensionLoader.getDefaultExtension();
+            }
+        }
+    }
+
+    @Override
+    public void setApplicationModel(ApplicationModel applicationModel) {
+        this.applicationModel = applicationModel;
+    }
+
+    @Override
+    public MetricsServiceExporter export() {
+        if (metricsService != null) {
+            if (!isExported()) {
+                ApplicationConfig applicationConfig = getApplicationConfig();
+                ServiceConfig<MetricsService> serviceConfig = new ServiceConfig<>();
+                serviceConfig.setScopeModel(applicationModel.getInternalModule());
+                serviceConfig.setApplication(applicationConfig);
+                serviceConfig.setRegistry(new RegistryConfig("N/A"));
+                serviceConfig.setProtocol(generateMetricsProtocol());
+                serviceConfig.setInterface(MetricsService.class);
+                serviceConfig.setDelay(0);
+                serviceConfig.setRef(metricsService);
+                serviceConfig.setGroup(applicationConfig.getName());
+                serviceConfig.setVersion(MetricsService.VERSION);
+
+                // export
+                serviceConfig.export();
```

Comment:
It's better to add the service config of `MetricsService` to internal module  instead of export it directly.
```java

applicationModel.getInternalModule().getConfigManager().addService(serviceConfig);

```